docs/source/internals/rest_api.rst

```diff
@@ -1,11 +1,298 @@
+
+.. code-block:: python
+
+    from aiida.restapi.api import AiidaApi, App
+    from aiida.restapi.run_api import run_api
+    from flask_restful import Resource
+
+To start with, we import the base classes to be extended/employed: ``AiidaApi`` and ``App``.
+For simplicity, it is advisable to import the method ``run_api``, as it provides an interface
+to configure the Api, parse command-line arguments, and couple the two classes representing the Api
+and the App. However, you can refer to the documentation of
+`flask_restful <https://flask-restful.readthedocs.io/>`_ to configure and hook-up an
+Api through its built-in methods.
+
+Then we define a class representing the additional resource:
+
+.. code-block:: python
+
+    class NewResource(Resource):
+        """
+        resource containing GET and POST methods. Description of each method
+        follows:
+
+        GET: returns id, ctime, and attributes of the latest created Dict.
+
+        POST: creates a Dict object, stores it in the database,
+        and returns its newly assigned id.
+
+        """
+
+        def get(self):
+            from aiida.orm import QueryBuilder, Dict
+
+            qb = QueryBuilder()
+            qb.append(Dict,
+                      project=['id', 'ctime', 'attributes'],
+                      tag='pdata')
+            qb.order_by({'pdata': {'ctime': "desc"}})
+            result = qb.first()
+
+            # Results are returned as a dictionary, datetime objects is
+            # serialized as ISO 8601
+            return dict(id=result[0],
+                        ctime=result[1].isoformat(),
+                        attributes=result[2])
+
+        def post(self):
+            from aiida.orm import Dict
+
+            params = dict(property1="spam", property2="egg")
+            paramsData = Dict(dict=params).store()
+
+            return {'id': paramsData.pk}
+
+The class ``NewResource`` contains two methods: ``get`` and ``post``.
+The names chosen for these functions are not arbitrary but fixed
+by ``Flask`` to individuate the functions that respond to HTTP request
+of type GET and POST, respectively.
+In other words, when the API receives a GET (POST) request to the
+URL ``new-endpoint``, the function ``NewResource.get()`` (``NewResource.post()``)
+will be executed.
+The HTTP response is constructed around the data returned by these functions.
+The data, which are packed as dictionaries, are serialized by Flask as a JSON
+stream of data. All the Python built-in types can be serialized by Flask
+(e.g. ``int``, ``float``, ``str``, etc.), whereas for serialization of custom types
+we let you refer to the `Flask documentation <http://flask.pocoo.org/docs/>`_ .
+The documentation of Flask is the main source of information also for topics
+such as customization of HTTP responses, construction of custom URLs
+(e.g. accepting parameters), and more advanced serialization issues.
+
+Whenever you face the need to handle errors, consider to use the AiiDA
```

Comment:
Can you fix also below, to put 1 sentence per line?

Reply:
Done!